src/app/child-dev-project/children/child-select/child-select.component.ts

```diff
@@ -21,56 +22,41 @@ import { MatAutocompleteTrigger } from "@angular/material/autocomplete";
 export class ChildSelectComponent implements OnChanges {
   searchText = "";
   showOnlyActiveChildren: boolean = true;
-  suggestions = new Array<Child>();
-  allChildren = new Array<Child>();
-  selectedChildren = new Array<Child>();
+  suggestions: Child[] = [];
+  allChildren: Child[] = [];
+  selectedChildren: Child[] = [];
 
-  @Input() valueAsIds: string[];
+  @Input() selectedChildrenIds: string[] = [];
+  @Output() selectedChildrenIdsChange = new EventEmitter<string[]>();
   @Input() disabled: boolean;
-  @Output() valueAsIdsChange = new EventEmitter();
-  @Output() newIdAdded = new EventEmitter();
-  @Output() idRemoved = new EventEmitter();
+  @Output() newIdAdded = new EventEmitter<string>();
+  @Output() idRemoved = new EventEmitter<string>();
 
-  @ViewChild("inputField", { static: true }) inputField;
+  @ViewChild("inputField", { static: true })
+  inputField: ElementRef<HTMLInputElement>;
   @ViewChild(MatAutocompleteTrigger) autocomplete: MatAutocompleteTrigger;
 
   constructor(private childrenService: ChildrenService) {}
 
   ngOnChanges(changes: SimpleChanges): void {
-    if (changes.hasOwnProperty("valueAsIds")) {
-      this.loadInitial();
+    if (changes.hasOwnProperty("selectedChildrenIds")) {
+      this.childrenService
+        .getChildren()
+        .pipe(untilDestroyed(this))
+        .subscribe((children) => {
+          this.allChildren = [];
```

Comment:
is `allChildren` ever getting filled?

Reply:
ah, I think I see how this is working now, after reading it several times ... We should make the code a bit more explicit and "standard" during #136

Reply:
Yes its because of the ngOnChanges method which might be called multiple times. I ran into a problem when I used this in the ChildSelectCompoment.